pkg/frontegg/sso_config.go

```diff
@@ -0,0 +1,205 @@
+package frontegg
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/MaterializeInc/terraform-provider-materialize/pkg/clients"
+)
+
+const (
+	SSOConfigurationsApiPathV1 = "/frontegg/team/resources/sso/v1/configurations"
+)
+
+// SSOConfig represents the structure for SSO configuration.
+type SSOConfig struct {
+	Id                string    `json:"id"`
+	Enabled           bool      `json:"enabled"`
+	SsoEndpoint       string    `json:"ssoEndpoint"`
+	PublicCertificate string    `json:"publicCertificate"`
+	SignRequest       bool      `json:"signRequest"`
+	AcsUrl            string    `json:"acsUrl"`
+	SpEntityId        string    `json:"spEntityId"`
+	Type              string    `json:"type"`
+	OidcClientId      string    `json:"oidcClientId"`
+	OidcSecret        string    `json:"oidcSecret"`
+	CreatedAt         time.Time `json:"createdAt"`
+	Domains           []Domain
+}
+
+// Domain represents the structure for SSO domain.
+type Domain struct {
+	ID          string `json:"id"`
+	Domain      string `json:"domain"`
+	Validated   bool   `json:"validated"`
+	SsoConfigId string `json:"sso_config_id"`
```

Comment:
To double check, this is intentionally snake-cased, right? The parent struct looks to be camelcased.